File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.py
class Solution(object):
    def numberOfBeams(self, bank):
        ans, count, prev = 0, 0, 0
        for s in bank:
            count = 0
            for c in s:
                if c == '1': count += 1
            if count != 0:
                ans += (prev * count)
                prev = count
        
        return ans

        

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int ans = 0, count = 0, prev = 0;
        for(string s : bank) {
            count = 0;
            for(char c : s) {
                if(c == '1') count++;
            }
            if(count != 0) {
                ans += prev * count;
                prev = count;
            }
        }

        return ans;
    }
};